dev-tools/packaging/packages.yml

```diff
@@ -16,6 +16,158 @@ shared:
+        source: '{{ repo.RootDir }}/LICENSE.txt'
+        mode: 0644
+      /Library/Application Support/{{.BeatVendor}}/{{.BeatName}}/README.md:
+        template: '{{ elastic_beats_dir }}/dev-tools/packaging/templates/common/README.md.tmpl'
+        mode: 0644
+      /Library/Application Support/{{.BeatVendor}}/{{.BeatName}}/.build_hash.txt:
+        content: >
+          {{ commit }}
+        mode: 0644
+      /Library/Application Support/{{.BeatVendor}}/{{.BeatName}}/{{.identifier}}.plist:
+        template: '{{ elastic_beats_dir }}/dev-tools/packaging/templates/darwin/launchd-daemon.plist.tmpl'
+        mode: 0644
+      /etc/{{.BeatName}}/agent.reference.yml:
+        source: 'agent.reference.yml'
+        mode: 0644
+      /etc/{{.BeatName}}/agent.yml:
+        source: 'agent.yml'
+        mode: 0600
+        config: true
+      /etc/{{.BeatName}}/beats/filebeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz:
+        source: '{{ elastic_beats_dir }}/x-pack/filebeat/build/distributions/filebeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz'
+        mode: 0644
+      /etc/{{.BeatName}}/beats/metricbeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz:
+        source: '{{ elastic_beats_dir }}/x-pack/metricbeat/build/distributions/metricbeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz'
+        mode: 0644
+
+  - &agent_binary_files
+    '{{.BeatName}}{{.BinaryExt}}':
+      source: build/golang-crossbuild/{{.BeatName}}-{{.GOOS}}-{{.Platform.Arch}}{{.BinaryExt}}
+      mode: 0755
+    LICENSE.txt:
+      source: '{{ repo.RootDir }}/LICENSE.txt'
+      mode: 0644
+    README.md:
+      template: '{{ elastic_beats_dir }}/dev-tools/packaging/templates/common/README.md.tmpl'
+      mode: 0644
+    .build_hash.txt:
+      content: >
+        {{ commit }}
+      mode: 0644
+    'agent.reference.yml':
+      source: 'agent.reference.yml'
+      mode: 0644
+    'agent.yml':
+      source: 'agent.yml'
+      mode: 0600
+      config: true
+
+  # Binary package spec (tar.gz for linux/darwin) for community beats.
+  - &agent_binary_spec
+    <<: *common
+    files:
+      <<: *agent_binary_files
+      'beats/filebeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz':
+        source: '{{ elastic_beats_dir }}/x-pack/filebeat/build/distributions/filebeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz'
+        mode: 0644
+      'beats/metricbeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz':
+        source: '{{ elastic_beats_dir }}/x-pack/metricbeat/build/distributions/metricbeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz'
+        mode: 0644
+
+  # Binary package spec (zip for windows) for community beats.
+  - &agent_windows_binary_spec
+    <<: *common
+    files:
+      <<: *agent_binary_files
+      install-service-{{.BeatName}}.ps1:
+        template: '{{ elastic_beats_dir }}/dev-tools/packaging/templates/windows/install-service.ps1.tmpl'
```

Comment:
Does the agent use the same CLI arguments as the beats? Maybe you need a customized install script?

Reply:
we tried to mimic beat behavior here and we support args like `-c` or `path.*` i'm not sure about `-E`